acme/http.go

```diff
@@ -36,8 +33,55 @@ const (
 
 	// ourUserAgent is the User-Agent of this underlying library package.
 	ourUserAgent = "xenolf-acme"
+
+	// caCertificateEnvVar is the environment variable name that can be used to
+	// specify the path to a PEM encoded CA Certificate that can be used to
+	// authenticate an ACME server with a HTTPS certificate not issued by a CA in
+	// the system-wide trusted root list.
+	caCertificateEnvVar = "CA_CERTIFICATE"
 )
 
+// init populates the default HTTPClient instance. The httpClient's Transport's
+// TLSClientConfig is populated with initCertPool to allow developers to
+// override the system trust store for interacting with test ACME servers.
+func init() {
+	HTTPClient = http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			Dial: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).Dial,
+			TLSHandshakeTimeout:   15 * time.Second,
+			ResponseHeaderTimeout: 15 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+			TLSClientConfig: &tls.Config{
+				RootCAs: initCertPool(),
+			},
+		},
+	}
+}
+
+// initCertPool creates a *x509.CertPool populated with the PEM certificate
+// found in the filepath specified in the caCertificateEnvVar OS environment
+// variable. If the caCertificateEnvVar is not set, or there is an error
+// creating a *x509.CertPool from the provided value then nil is returned and
+// the default Golang CertPool is used.
+func initCertPool() *x509.CertPool {
```

Comment:
My approach to error handling with this function was to fall-back to the system certpool by returning `nil`. This means providing an invalid env var arg will silently fall back to the system trust store and probably produce an HTTPS validation error on first connect to the ACME server. That seemed like the best way forward but if you would rather see a `panic` or a warning message logged that would be fine with me :-)

Reply:
Considering that the developer specifically provides the environment variable to add a CA, I would honestly rather see it panic/fail when loading/parsing of the designated file fails. The error you'd get would just say "not trusted", but it wouldn't be clear that it was because loading the file failed.

Reply:
Yeah, I think an error must always at least be logged. I think it should be returned or logged (or a panic, although I'm less fond about panics due to env var misconfigurations).

Reply:
There's a few different options thrown out in this thread. Is there one that everyone would be happy with?

It sounds like @mholt would like to avoid a panic but always have something logged while @wyattjoh , it sounds like you want to see panic/fail. Would using `log.Fatal` in the failure cases of `initCertPool` be acceptable to both of you?

> I think an error must always at least be logged. I think it should be returned or logged

@mholt If `initCertPool` returns an error I think I'll have to undo ee12bff and put the initialization of the `HTTPClient` somewhere where I can unpack the result and potential error from `initCertPool` before assigning the `TLSClientConfig`'s `RootCAs`.

>  (or a panic, although I'm less fond about panics due to env var misconfigurations).

Speaking of env var misconfigurations, it also occurred to me that maybe using `CA_CERTIFICATES` as the env var name is too generic and more likely to collide with other software. I think it would be sensible to change that to `LEGO_CA_CERTIFICATES` while we're iterating on this branch anyway.

Reply:
+1 on the more specific env var. In that case, a panic is probably fine, since an environment variable named like that which was set wrong would definitely not be an accident.

Reply:
Ok, sounds like a more specific env var and a `panic` would satisfy everyone. :+1: